subsys/bluetooth/host/addr.c

```diff
@@ -120,3 +120,19 @@ bool bt_addr_le_is_resolved(const bt_addr_le_t *addr)
 {
 	return (addr->type & ADDR_RESOLVED_BITMASK) != 0;
 }
+
+bool bt_addr_le_is_valid(const bt_addr_le_t *addr)
+{
+	if (addr == NULL) {
+		return false;
+	}
+
+	if (addr->type != BT_ADDR_LE_PUBLIC && addr->type != BT_ADDR_LE_RANDOM &&
+	    addr->type != BT_ADDR_LE_PUBLIC_ID && addr->type != BT_ADDR_LE_RANDOM_ID) {
```

Comment:
These are not valid Bluetooth LE address types: `BT_ADDR_LE_PUBLIC_ID` `BT_ADDR_LE_RANDOM_ID`.

Rather, those have a bit set that informs the host that the controller saw a RPA on air, but the controller resolved it for us.